src/allmydata/scripts/debug.py

```diff
@@ -933,34 +928,34 @@ def flip_bit(start, end):
         f.write(d)
         f.close()
 
-    f = open(fn, "rb")
-    prefix = f.read(32)
-    f.close()
-    if prefix == MutableShareFile.MAGIC:
-        # mutable
-        m = MutableShareFile(fn)
-        f = open(fn, "rb")
-        f.seek(m.DATA_OFFSET)
-        data = f.read(2000)
-        # make sure this slot contains an SMDF share
-        assert data[0:1] == b"\x00", "non-SDMF mutable shares not supported"
-        f.close()
+    with open(fn, "rb") as f:
+        prefix = f.read(32)
 
-        (version, ig_seqnum, ig_roothash, ig_IV, ig_k, ig_N, ig_segsize,
-         ig_datalen, offsets) = unpack_header(data)
-
-        assert version == 0, "we only handle v0 SDMF files"
-        start = m.DATA_OFFSET + offsets["share_data"]
-        end = m.DATA_OFFSET + offsets["enc_privkey"]
-        flip_bit(start, end)
-    else:
-        # otherwise assume it's immutable
-        f = ShareFile(fn)
-        bp = ReadBucketProxy(None, None, '')
-        offsets = bp._parse_offsets(f.read_share_data(0, 0x24))
-        start = f._data_offset + offsets["data"]
-        end = f._data_offset + offsets["plaintext_hash_tree"]
-        flip_bit(start, end)
+        if MutableShareFile.is_valid_header(prefix):
+            # mutable
+            m = MutableShareFile(fn)
+            f = open(fn, "rb")
```

Comment:
existing code i think, but wrapping this block in "with open(...)" would be nice too